lib/topics_filter.rb

```diff
@@ -474,9 +479,37 @@ def filter_created_by_user(usernames:)
       )
   end
 
+  def apply_custom_filter!(scope:, filter_name:, values:)
+    values.dup.each do |value|
+      custom_key = "#{filter_name}:#{value}"
+      if custom_match =
+           DiscoursePluginRegistry.custom_filter_mappings.find { |hash| hash.key?(custom_key) }
+        scope = custom_match[custom_key].call(scope, custom_key, @guardian) || scope
+        values.delete(value)
+      end
+    end
+    scope
+  end
+
+  def ensure_topic_users_reference!
+    if @guardian.authenticated?
+      if !@loaded_topic_users_reference
+        @scope =
+          @scope.joins(
+            "LEFT JOIN topic_users tu ON tu.topic_id = topics.id
+            AND tu.user_id = #{@guardian.user.id.to_i}",
```

Comment:
```suggestion
            AND tu.user_id = #{@guardian.user.id}",
```

Not sure why `to_i` is needed, it should already be `int`